### ISSUE TITLE
ltcencode and ltcdecode unexpected results

### ISSUE THREAD

**Issue by samgardner** [x42/libltc#15]
Hi,
I've just installed libltc and run into an issue that I'm not sure is the correct behaviour:

    # ./ltcencode test
    sample rate: 48000.00
    frames/sec: 25.00
    secs to write: 2.00
    sample format: 8bit unsigned mono
    Done: wrote 96000 samples to 'test'
    # ./ltcdecode test
    * reading from: test

Should the results of ltcencode be readable by ltcdecode? I'd expect so. I've also tried playing with the decode sample rate but receive the same results.

I've been getting funny results with ltcdecode from other audio sources (MOTU micro express, android LTC generator), but thought I'd check the simplest case with you first. 
Server is running CentOS 6.6 2.6.32-431.el6.x86_64

**Comment by x42** [comment]
Yes, though for the simple `ltcencode`, `ltcdecode` have hardcoded  audio-frames per video-frames. Look at  https://github.com/x42/ltc-tools for a fully featured decoder.

Here's the output of `make check`
```
-----------------------------------------------------------------
./ltcencode output.raw
sample rate: 48000.00
frames/sec: 25.00
secs to write: 2.00
sample format: 8bit unsigned mono
Done: wrote 96000 samples to 'output.raw'
./ltcdecode output.raw | diff -q ./expect_48k_2sec.txt -
* reading from: output.raw
-----------------------------------------------------------------
./ltcencode output.raw 192000
sample rate: 192000.00
frames/sec: 25.00
secs to write: 2.00
sample format: 8bit unsigned mono
Done: wrote 384000 samples to 'output.raw'
./ltcdecode output.raw 7680 | diff -q ./expect_96k_2sec.txt -
* reading from: output.raw
-----------------------------------------------------------------
./ltcdecode ./timecode.raw 882 | diff -q ./timecode.txt -
* reading from: ./timecode.raw
-----------------------------------------------------------------
  libltc-1.1.4 passed all tests.
-----------------------------------------------------------------
```

Does it match on your machine?

**Comment by samgardner** [comment]
-----------------------------------------------------------------
    ./ltcencode output.raw
    sample rate: 48000.00
    frames/sec: 25.00
    secs to write: 2.00
    sample format: 8bit unsigned mono
    Done: wrote 96000 samples to 'output.raw'
    ./ltcdecode output.raw | diff -q ./expect_48k_2sec.txt -
    * reading from: output.raw
    Files ./expect_48k_2sec.txt and - differ
    make[1]: *** [check] Error 1
    make[1]: Leaving directory `/home/ingest/dev/libltc/tests'
    make: *** [check-recursive] Error 1

This is on a clean repo straight from github.

**Comment by samgardner** [comment]
I've just installed the repo on a different machine - Debian 7.8 running 3.2.65-1+deb7u2, and it passes all the tests. Ideally I'd need the CentOS machine working with this lib. Are there any config files you would like me to supply to help fix this? Perhaps something is amiss in the autotools step?

**Comment by x42** [comment]
I cannot reproduce. Can you paste the complete output of 
```
cd /tmp
git clone git://github.com/x42/libltc.git
cd libltc
./autogen.sh
make check
```

Here it finished with "passed all tests.".

I don't know CentOS, but it's Fedora based, isn't it? First google-hit for "libtltc rpm build log" shows a successful self-test: http://arm.koji.fedoraproject.org//packages/libltc/1.1.4/4.fc23/data/logs/aarch64/build.log

**Comment by samgardner** [comment]
Here's the complete output of those steps: http://pastebin.com/DzQDinFw

In the meantime I'll dig around elsewhere for other packages, if I find any solutions I'll share them here. Thanks!

**Comment by samgardner** [comment]
Just an update, unfortunately there are no supported packages for CentOS 6 (to my knowledge). A few exist for CentOS 7 but I'm stuck with 6 due to other applications on this server.

I've just tried installing the src rpm's from fedoraproject.org and still fail the tests. A dump from installing libltc-1.1.2-1.fc18.src.rpm is here: http://pastebin.com/4zA9a6PW
Installation is invoked via rpmbuild --rebuild libltc-1.1.2-1.fc18.src.rpm

**Comment by samgardner** [comment]
I just tried installing libltcsmpte-0.4.4 and it seems to have successfully installed and passed all the tests. It even seems to be behaving with my external audio sources.

For the time being I'll use libltcsmpte.

**Comment by x42** [comment]
whatever works, I suppose.

Still it'd be nice to get to the bottom of this issue, sadly I can't see a clue as to why it fails on CentOS in the build-log.  

What gcc version comes with centOS 6? 
Best guess is that the byte-order detection in `ltc.h` fails with old gcc.

**Comment by x42** [comment]
oops, didn't mean to close this.

**Comment by samgardner** [comment]
# gcc -v 
    Using built-in specs.
    Target: x86_64-redhat-linux
    Configured with: ../configure --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-bootstrap --enable-shared --enable-threads=posix --enable-checking=release --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-gnu-unique-object --enable-languages=c,c++,objc,obj-c++,java,fortran,ada --enable-java-awt=gtk --disable-dssi --with-java-home=/usr/lib/jvm/java-1.5.0-gcj-1.5.0.0/jre --enable-libgcj-multifile --enable-java-maintainer-mode --with-ecj-jar=/usr/share/java/eclipse-ecj.jar --disable-libjava-multilib --with-ppl --with-cloog --with-tune=generic --with-arch_32=i686 --build=x86_64-redhat-linux
    Thread model: posix
    gcc version 4.4.7 20120313 (Red Hat 4.4.7-11) (GCC)

**Comment by x42** [comment]
I was able to reproduce this with gcc 4.4.7.  the problem was endianess detection. fixed now in 5dc71c37c65506
